akka-persistence/src/main/scala/akka/persistence/serialization/MessageSerializer.scala

```diff
@@ -209,14 +213,21 @@ class MessageSerializer(val system: ExtendedActorSystem) extends BaseSerializer
     AtomicWrite(atomicWrite.getPayloadList.asScala.map(persistent)(collection.breakOut))
   }
 
-  private def payload(persistentPayload: mf.PersistentPayload): Any = {
+  protected def payload(persistentPayload: mf.PersistentPayload): Any = {
     val manifest = if (persistentPayload.hasPayloadManifest)
-      persistentPayload.getPayloadManifest.toStringUtf8 else ""
-
-    serialization.deserialize(
-      persistentPayload.getPayload.toByteArray,
-      persistentPayload.getSerializerId,
-      manifest).get
+      persistentPayload.getPayloadManifest.toStringUtf8
+    else ""
+
+    if (deserializeUsingManifest) {
+      serialization.deserialize(
+        persistentPayload.getPayload.toByteArray,
+        Class.forName(manifest)).get
```

Comment:
a manifest doesn't have to be a class name, in fact we encourage to use something else than a class name so that the class can be renamed (see `SerializerWithStringManifest`)